### ISSUE TITLE
issue with autohide bookmarks and main toolbar in fullscreen

### ISSUE THREAD

**Issue by marcmy** [MrOtherGuy/firefox-csshacks#154]
hello, I know know nothing about css but decided to try some of these out. Very impressed so far but I've run into an issue with autohide_bookmarks_and_main_toolbars while in fullscreen mode. It hides and unhides as it should but there appears a black bar on the tabs part of the toolbar. Have no other scripts enabled and it doesn't seem to matter what theme or tab density is set.
[https://i.imgur.com/0GFAvSe.png](https://i.imgur.com/0GFAvSe.png) 
Using Firefox Nightly on Windows 11

**Comment by MrOtherGuy** [comment]
Thanks for reporting. It was pretty simple fix. The behavior looks kinda janky though if you have set toolbars to hide in fullscreen (using Firefox' own options) but it works at least.

**Comment by marcmy** [comment]
thanks, it's much better. only now the bookmarks toolbar has an invisible background, but i can live with that. :)

**Comment by MrOtherGuy** [comment]
What do you by invisible background? Is bookmarks toolbar partially visible somehow for you? Note that bookmarks toolbar isn't supposed to be visible at all in fullscreen mode because it isn't in Firefox normally, and this style shouldn't change that.

**Comment by marcmy** [comment]
ah, didn't know. but yeah, it's visible in fullscreen, i can see all the bookmarks but there's no background, as in it's transparent.

**Comment by MrOtherGuy** [comment]
Do you use some other css besides that style?

**Comment by marcmy** [comment]
currently just these 

@import url(chrome/tab_close_button_always_on_hover.css);
@import url(chrome/tab_loading_progress_throbber.css);
@import url(chrome/button_effect_scale_onhover.css);
@import url(chrome/button_effect_icon_glow.css);
@import url(chrome/autohide_bookmarks_and_main_toolbars.css);

**Comment by MrOtherGuy** [comment]
Hmm, I suppose those button effect styles might cause this. Could you test if disabling those makes bookmarks toolbar fully invisible, thanks? If those do actually cause the issue then it's definitely a problem and I need to fix them.

But if it's not them then I need to think about this later.

Also just to be sure, does the problem occur only in fullscreen mode?

**Comment by marcmy** [comment]
yup, tried with only the autohide_bookmarks_and_main_toolbars.css and still shows, only in fullscreen mode.

**Comment by MrOtherGuy** [comment]
Alright, thanks. I'll get back to you later.

**Comment by marcmy** [comment]
hey, just noticed another issue. the downloads button doesn't seem to work when set to "hide button when empty". after downloading a file the button appears but does nothing when i click it.

**Comment by MrOtherGuy** [comment]
> 

Hmmh. Unfortunately I cannot reproduce that. The downloads buttons seem to work perfectly fine when I apply your CSS setup. You should probably test if it works without any custom CSS

As for the original issue... sadly I don't think I can do anything without you doing some digging. I cannot reproduce that either, maybe there is some way I could reproduce the contents of bookmarks toolbar being visible, but I just don't know how to trigger it. I suppose it's *possible* it's somehow Win11 specific but I find that hard to believe.

What you could do to help me figure it out is to make Firefox fullscreen, and then hit `Ctrl+Shift+J`. It will open browser console, and then execute this snippet: `getComputedStyle(document.getElementById("PersonalToolbar")).visibility`

It should print "hidden"

Then execute this: `getComputedStyle(document.getElementById("PersonalToolbar")).display`

That should print "-moz-box"

If either prints anything else then something has gone wrong and I don't know what that might be.

**Comment by marcmy** [comment]
Alright, I suspected something might be wrong with my FF profile, so I created a fresh one, but unfortunately. It seems the issue is there as well. The download button issue along with the the bookmarks bar. I ran those snippets and they returned the same as you said. 

The only other thing I thought could possibly be conflicting was my taskbar. I have it set on auto-hide along with a program to make it transparent/acrylic, which creates a white line at the bottom of my screen only in Firefox. Disabled those two, but the issue remains. 

I will try migrating my profile to my laptop which is running Linux tomorrow, keep you posted.

**Comment by MrOtherGuy** [comment]
Did you try if the issue with downloads button happens without any css?

**Comment by marcmy** [comment]
Yes. The downloads button issue and the bookmarks toolbar in fullscreen both only happen with autohide_bookmarks_and_main_toolbars.css enabled, regardless of whatever else is enabled or not.

Another update, I tried it on my Linux machine as well with a brand new profile. Same issue.

**Comment by MrOtherGuy** [comment]
Interesting. At least on Linux I can try to reproduce the problem.

**Comment by MrOtherGuy** [comment]
Unfortunately I have not been able to reproduce the bookmarks being visible in fullscreen mode at all on my Linux (Fedora 35) system. I really don't understand what might make such a difference, because I tried both my existing profiles, as well as new profiles and I just couldn't reproduce it.

I was, however, able to reproduce the downloads button thing - but only once which is confusing. It seems the layout javascript logic checks if the button is visible and it might trip if it is not. But that's equally confusing, since then I would expect that I could reproduce it every time but that doesn't seem to be happening. It's not even only the first time after Firefox starts, because except for that one time, it always worked properly even after Firefox was restarted.

I'm afraid, there's little I can do here for the time being.

However, I have pushed an update to autohide_bookmarks_and_main_toolbars.css that fixes bunch of mostly unrelated stuff but it *might* potentially have some effects on why you see bookmarks in fullscreen mode.

**Comment by marcmy** [comment]
Hey.. so I've tried everything I could think of. Tried to think what I had in common with my Linux machine install. Dark mode theme? Nope. The chrome folder files? Tried deleting them and ended up just having only the autohide_bookmarks_and_main_toolbars.css in my chrome folder renamed to userChrome.css. Then finally decided to completely wipe Firefox from my system, data folders, registry entries and all, and reinstall it. All of that fixed the slow loading and updating issues I was having, but the bookmarks and downloads button issue remains. It's definitely not a big deal but I'm out of ideas, lol. It's odd that it persists for me across multiple installs/operating systems and even computers, yet not for you, but stranger things have happened. Anyway, just thought I'd let you know, and thanks for trying to solve it. Great work, by the way! Looking forward to trying some more of the tweaks.

**Comment by MrOtherGuy** [comment]
Well that's unfortunate to say the least. Could you maybe attach a screenshot of how the bookmarks toolbar looks when it glitches? Preferrably the full window size. It's a long shot, but maybe it would provide some clues.

**Comment by marcmy** [comment]
Sure! here you go:

![EW0lB9n 1](https://user-images.githubusercontent.com/21000174/141791741-ba3887c4-398e-44ca-bba0-4da1f51b67cf.png)

**Comment by MrOtherGuy** [comment]
Hmm that is odd. I'm wondering if this actually might be some graphics driver or otherwise rendering problem. If you look at the area between your bookmarks "b550" and "x570" there's pretty apparent discontinuity there - and more generally it doesn't look like the bookmarks toolbar area is actually transparent in that you see the page contents through it. You see *something* behind it but not really what's supposed to be there.

I suppose you could try and see what happens if you disable hardware acceleration

**Comment by marcmy** [comment]
Hmm, interesting. Hadn't really noticed any discontinuity. Also it's strange that it's not transparent in that screenshot. It is actually when I'm looking at it in fullscreen though, but maybe that's cause I'm on my Linux install atm. 

Major update while writing this, I haven't quite figured out how to make clickable links launch Nightly in Linux instead of the default browser (which is based on stable Firefox), but I went ahead and copied over the profile with the css customizations, and lo and behold, the issue wasn't there. I installed stable Firefox to test that as well and it's not there either. So I can pretty much confirm that this is Nightly-only.

**Comment by MrOtherGuy** [comment]
Wow, this is in fact something that only happens in Nightly and as it turns out I never tested the style on nightly even if I usually do.

I don't know what causes that though, all the styling suggests it should be the same, so that's bit of a mystery.

Regardless, I have added a simple fix for this in d35ebf1 which seems to work fine.

Thanks for all the info about this, I would have never caught this otherwise.

**Comment by marcmy** [comment]
You got it! Glad I could help and thanks again :)

**Comment by marcmy** [comment]
Edit: Afraid the downloads button still isn't working :( Also works on stable but not nightly.

**Comment by MrOtherGuy** [comment]
Sorry, I didn't explain this before - but there's nothing you can do with CSS to fix that.

Your options are basically to move downloads button into tabs toolbar (since that isn't hidden in your setup), or to overflow menu, or set a pref `browser.download.alwaysOpenPanel` to `false` so Firefox doesn't try to open the downloads panel - the whole issue is because it tries to open it and fails because the button is not available.
